webapp/store/posts.js

```diff
@@ -0,0 +1,77 @@
+import gql from 'graphql-tag'
+
+export const state = () => {
+  return {
+    posts: [],
+  }
+}
+
+export const mutations = {
+  SET_POSTS(state, posts) {
+    state.posts = posts || null
+  },
+}
+
+export const getters = {
+  posts(state) {
+    return state.posts || []
+  },
+}
+
+export const actions = {
+  async fetchPosts({ commit, dispatch }, { i18n, filter }) {
+    const client = this.app.apolloProvider.defaultClient
+    const {
+      data: { Post },
+    } = await client.query({
+      query: gql(`
+        query Post($filter: _PostFilter, $first: Int, $offset: Int) {
+          Post(filter: $filter, first: $first, offset: $offset) {
+            id
+            title
+            contentExcerpt
+            createdAt
+            disabled
+            deleted
+            slug
+            image
+            author {
+              id
+              avatar
+              slug
+              name
+              disabled
+              deleted
+              contributionsCount
+              shoutedCount
+              commentsCount
+              followedByCount
+              followedByCurrentUser
+              location {
+                name: name${i18n.locale().toUpperCase()}
+              }
+              badges {
+                id
+                key
+                icon
+              }
+            }
+            commentsCount
+            categories {
+              id
+              name
+              icon
+            }
+            shoutedCount
+          }
+      }`),
+      variables: {
+        filter,
+        first: 12,
+        offset: 0,
+      },
+    })
+    commit('SET_POSTS', Post)
+    return Post
```

Comment:
would be nice to use `filterPosts` from `PostQuery`, but when I tried it, it's not returning the proper query, but a `Document` object... I think because you need to use `return filterPosts(i18n)`,but not sure how to do that here :thinking: